core/src/OpenMPTarget/Kokkos_OpenMPTarget_Instance.cpp

```diff
@@ -67,29 +67,21 @@ void OpenMPTargetInternal::fence(const std::string& name,
   }
 }
 int OpenMPTargetInternal::concurrency() const {
-  int max_threads = 2048 * 80;
-#if defined(KOKKOS_IMPL_ARCH_NVIDIA_GPU)
   int max_threads_sm = 2048;
+  int max_threads    = max_threads_sm * 80;
 #if defined(KOKKOS_ARCH_AMPERE86)
   max_threads = max_threads_sm * 84;
 #elif defined(KOKKOS_ARCH_AMPERE87)
   max_threads_sm = 1024;
   max_threads    = max_threads_sm * 32;  // Orin Nano cores
 #elif defined(KOKKOS_ARCH_AMPERE80)
-  max_threads = max_threads_sm * 108;
+  return max_threads_sm * 108;
 #elif defined(KOKKOS_ARCH_VOLTA72)
-  max_threads = max_threads_sm * 84;
+  return max_threads_sm * 84;
 #elif defined(KOKKOS_ARCH_VOLTA70)
-  max_threads = max_threads_sm * 80;
+  return max_threads_sm * 80;
 #elif defined(KOKKOS_ARCH_PASCAL60) || defined(KOKKOS_ARCH_PASCAL61)
-  max_threads = max_threads_sm * 60;
-#endif
-#elif defined(KOKKOS_ARCH_INTEL_GPU)
-#pragma omp target map(max_threads)
-  { max_threads = omp_get_num_procs(); }
-
-  // Multiply the number of processors with the SIMD length.
-  max_threads *= 32;
+  return max_threads_sm * 60;
 #endif
 
   return max_threads;
```

Comment:
This is kind of a mess.  Could we revisit it a little bit to increase readability/maintainability?

Reply:
ok